src/pymor/reductors/coercive.py

```diff
@@ -192,13 +217,39 @@ def append_vector(U, R, RR):
 
         estimator_matrix = NumpyMatrixOperator(estimator_matrix)
 
-        error_estimator = SimpleCoerciveRBEstimator(estimator_matrix, self.coercivity_estimator)
+        # optional output estimate
+        if self.assemble_output_error_estimate and self.fom.output_functional.linear:
+            output_estimator_matrices, output_functional_coeffs = self.assemble_output_error_estimator()
+        else:
+            output_estimator_matrices = output_functional_coeffs = None
+
+        error_estimator = SimpleCoerciveRBEstimator(estimator_matrix, self.coercivity_estimator,
+                                                    output_estimator_matrices, output_functional_coeffs)
         self.extends = (len(RB), dict(R_R=R_R, RR_R=RR_R, R_Os=R_Os, RR_Os=RR_Os))
 
         return error_estimator
 
+    def assemble_output_error_estimator(self):
+        output_estimator_matrices, output_functional_coeffs = [], []
+        output_func = self.fom.output_functional
+        product = self.products['RB']
+        if not isinstance(output_func, LincombOperator):
```

Comment:
Somewhere one needs to check that `output_func.operators` consists of linear, non-parametric `Operators`.

Reply:
Ah you are right for the non parametric part.
The linear part is already inherited in the `self.fom.output_functional.linear` check, I would hope.